plugins/BEdita/Core/src/Model/Action/SetAssociatedAction.php

```diff
@@ -123,15 +123,30 @@ protected function belongsTo(EntityInterface $entity, EntityInterface $relatedEn
 
         if ($existing === null && $relatedEntity === null) {
             return 0;
-        } elseif ($relatedEntity !== null) {
+        }
+        $joinData = [];
+        if ($relatedEntity !== null) {
             $bindingKey = (array)$this->Association->getBindingKey();
-
-            if ($existing !== null && $relatedEntity->extract($bindingKey) == $existing->extract($bindingKey)) {
+            // We use `_joinData` between Tree and Folder in `ParentObjects` association
+            // if '_joinData` are not set (common case) or not changed save is not performed
+            $joinData = (array)$relatedEntity->get('_joinData');
+            $diff = array_diff_assoc(
+                $joinData,
+                $entity->extract(array_keys($joinData))
+            );
+
+            if (
+                $existing !== null &&
+                $relatedEntity->extract($bindingKey) == $existing->extract($bindingKey) &&
+                empty($diff)
+            ) {
                 return 0;
             }
         }
 
         $entity->set($this->Association->getProperty(), $relatedEntity);
+        // set join data anyway, if empty no properties are set
+        $entity->set($joinData);
```

Comment:
Since `belongsTo` association doesn't have a join table I think that handling it in this way could bring to unwanted behavior if the Action was used on another `belongsTo` relation (not between `Tree` and `Folder`).

I think that the right place to setup the data on `$entity` should be `UpdateRelatedObjectsAction::prepareData()` https://github.com/bedita/bedita/blob/4-cactus/plugins/BEdita/Core/src/Model/Action/UpdateRelatedObjectsAction.php#L69

There we could add joinData to `$entity`.

Anyway then we would find a way to force the save of  `$entity` in `belongsTo()` method. 
This could be done for example setting a runtime config `$this->setConfig('forceUpdate', true);` always in `UpdateRelatedObjectsAction::prepareData()` and reading it in `SetAssociatedAction::belongsTo()` or maybe checking `$entity->isDirty()` as first thing in `SetAssociatedAction::belongsTo()` (not sure of that)

Reply:
Ok, seems quite clear.
I'll try that solution with `->isDirty()` as flag